src/histolab/filters/image_filters_functional.py

```diff
@@ -0,0 +1,744 @@
+    """Compute the Hysteresis threshold on the complement of an image in greyscale and
+    return boolean mask based on pixels above this threshold.
+
+    Parameters
+    ----------
+    img : PIL.Image.Image
+        Input image.
+    low : int, optional (default is 50)
+        low threshold
+    high : int, optional (default is 100)
+         high threshold
+
+    Returns
+    -------
+    np.ndarray
+     Boolean NumPy array where True represents a pixel above Otsu threshold.
+    """
+    gs = ImageOps.grayscale(img)
+    comp = invert(gs)
+    hyst_mask = sk_filters.apply_hysteresis_threshold(np.array(comp), low, high)
+    return hyst_mask
+
+
+def otsu_threshold(img: PIL.Image.Image) -> np.ndarray:
+    """
+    Compute Otsu threshold on image as a NumPy array and return boolean mask
+    based on pixels above this threshold.
+
+    Note that Otsu threshold is expected to work correctly only for grayscale images
+
+    Parameters
+    ----------
+    img : PIL.Image.Image
+        Input image.
+
+    Returns
+    -------
+    np.ndarray
+     Boolean NumPy array where True represents a pixel above Otsu threshold.
+    """
+    img_arr = np.array(img)
+    otsu_thresh = sk_filters.threshold_otsu(img_arr)
+    # TODO UserWarning: threshold_otsu is expected to work correctly only for grayscale images
+    return _filter_threshold(img_arr, otsu_thresh)
+
+
+def local_otsu_threshold(img: PIL.Image.Image, disk_size: int = 3) -> np.ndarray:
+    """Compute local Otsu threshold for each pixel and boolean mask
+    based on pixels being less than the local Otsu threshold.
+
+    Note that the input image must be 2D.
+
+    Parameters
+    ----------
+    img: PIL.Image.Image
+        Input 2-dimensional image
+    disk_size :int, optional (default is 3)
+        Radius of the disk structuring element used to compute
+        the Otsu threshold for each pixel.
+
+    Returns
+    -------
+    np.ndarray
+        NumPy boolean array representing the mask based on local Otsu threshold
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    local_otsu = sk_filters.rank.otsu(img_arr, sk_morphology.disk(disk_size))
+    return local_otsu
+
+
+def filter_entropy(
+    img: PIL.Image.Image, neighborhood: int = 9, threshold: int = 5
+) -> np.ndarray:
+    """Filter image based on entropy (complexity).
+
+    Note that input must be 2D.
+
+    Parameters:
+    ----------
+    img: PIL.Image.Image
+        input 2-dimensional image
+    neighborhood : int, optionale (default is 9)
+        Neighborhood size (defines height and width of 2D array of 1's).
+    threshold: Threshold value.
+
+    Returns
+    -------
+    np.ndarray
+        NumPy boolean array where True represent a measure of complexity.
+    """
+    if img.size != 2:
+        raise ValueError("Input must be 2D.")
+    img_arr = np.array(img)
+    entropy = sk_filters.rank.entropy(img_arr, np.ones((neighborhood, neighborhood)))
```

Comment:
not sure but is there a wisest way to refactor the `np.ones` statement?

Reply:
could'n find an easiest way to define a square matrix without specifying both dimensions